core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/JDBCSourceAsColumnarStore.scala

```diff
@@ -243,11 +240,8 @@ class JDBCSourceAsColumnarStore(private var _connProperties: ConnectionPropertie
   def closeConnection(c: Option[Connection]): Unit = {
     c match {
       case Some(conn) if !conn.isClosed =>
-        conn match {
-          case ConnectionType.Embedded =>
-            // conn commit removed txState from the conn context.
-            conn.commit()
-            conn.close()
+        connectionType match {
+          case ConnectionType.Embedded => // Do nothing
```

Comment:
Ok. @suranjan Can you take a look at this change to see if skipping close here is fine, or should it be fixed in some other way. It is being invoked from ColumnInsertExec generated so please check if this could cause pool to be exhausted or something.